cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -160,6 +160,9 @@ func Run(s *options.CMServer) error {
 	if err != nil {
 		return err
 	}
+	// it is more safe to use hostname plus pod name
+	// see https://github.com/kubernetes/kubernetes/issues/22775
+	id = id + "-" + os.Getenv("POD_NAME")
```

Comment:
Originally, the hostname is not the pod name, since `hostNetwork` is set to be true, from my test via vagrant provider, it shows to be `node-1` which could not be the pod name, could you verify again?